crypto/src/tests/bls12381_tests.rs

```diff
@@ -400,3 +403,75 @@ async fn signature_service() {
     // Verify the signature we received.
     assert!(pk.verify(digest.as_ref(), &signature).is_ok());
 }
+
+// Checks if the private keys zeroed out
+#[test]
+fn test_zeroization_priv_key() {
+    let kp = keys().pop().unwrap();
+    let mut sk = kp.private();
+    let mut sk_bytes = Vec::new();
+    sk_bytes.extend_from_slice(sk.as_ref());
+
+    let ptr = std::ptr::addr_of!(sk.privkey) as *const u8;
+    let bytes_ptr = &sk.as_ref()[0] as *const u8;
+
+    unsafe {
+        let mut vec = Vec::new();
+        for i in 0..BLS12381PrivateKey::LENGTH {
+            vec.push(*bytes_ptr.add(i));
+        }
+        assert_eq!(&vec[..], &sk_bytes[..]);
+    }
+
+    sk.zeroize();
+
+    // Check that self.privkey is zeroized
+    unsafe {
+        for i in 0..BLS12381PrivateKey::LENGTH {
+            assert!(*ptr.add(i) == 0);
+        }
+    }
+
+    // Check that self.bytes is zeroized
+    unsafe {
+        let mut vec = Vec::new();
+        for i in 0..BLS12381PrivateKey::LENGTH {
+            vec.push(*bytes_ptr.add(i));
+        }
+        assert_ne!(&vec[..], &sk_bytes[..]);
+    }
+}
```

Comment:
YMMV, but something like 
`let memory: &[u8] = unsafe { ::std::slice::from_raw_parts(ptr, BLS12381PrivateKey::LENGTH) }` may just give you something easier to inspect, no?

Reply:
Thanks for this! I think it is much cleaner - I haven't touched unsafe rust much so I just didn't know about this.